enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,499 @@
+
+
+#### Egress Service
+
+Egress service functions similar to Egress IP in local gateway mode, with the exception that all traffic paths go through the kernel networking stack. Egress Service also uses IP rules and VRFs in order to match on traffic and forward it out the right network (if specified in the CRD). It uses iptables in order to SNAT packets to the load balancer IP. Like Egress IP, with unique networks there will need to be IP rules with higher precedence to match on packets from specific networks and direct them to the right VRF.
+
+
+#### Multiple External Gateways (MEG)
+
+There will be no support for MEG or pod direct ingress on any network other than the primary default network. Remember, MEG works the same way in local or shared gateway mode, by utilizing the shared gateway path. This support may be enhanced later by extending VRFs/networks outside of the cluster.
+
+
+### Host Networked Pods
+
+By encompassing VRFs into the host, this introduces some constraints and requirements for the behavior of host networked type pods. If a host networked pod is created in a Kubernetes namespace that has a unique network, should it be restricted to only communicating with that Kubernetes network? If the answer is yes, then we would need to do the following:
+
+With Linux VRFs, the application would need to bind directly to the VRF device in order to use TCP/UDP type sockets on the network. Raw sockets by default are allowed to span multiple VRFs. This seems like a problematic solution, as we cannot expect all user applications to provide configuration for which device to bind to for TCP/UDP sockets. There is sysctl configuration that will allow TCP/UDP sockets to listen across all VRFs, but that means then a host application could communicate with any network:
+
+
+```
+       sysctl -w net.ipv4.tcp_l3mdev_accept=1
+       sysctl -w net.ipv4.udp_l3mdev_accept=1
+```
+
+
+In order to restrict applications from being able to talk on the other networks, it seems like the only choice would be to insert iptables rules to block traffic coming from/to any of the network interfaces for that application. See [https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/Documentation/networking/vrf.rst?h=v6.1](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/Documentation/networking/vrf.rst?h=v6.1) for more details.
+
+From the OVN-Kubernetes perspective, we will support allowing any host networked pod to reach a Kubernetes service. This will work naturally as host network pods will be communicating with the service CIDR. For host network pod to pod direct access across the unique network, OVN-Kubernetes will only support this traffic when the application is bound to the VRF device. Modifying sysctl settings to allow applications to listen across all VRFs will be unsupported. However, Kubelet probes from the host to the a pod primary IP address on the default network will continue to function.
```

Comment:
> This will work naturally as host network pods will be communicating with the service CIDR.

I don't understand what that means.

Reply:
I think I've clarified it and added another section specifically talking about host networked pods -> services. Let me know if that makes sense.